cpp/modules/luma.gl/webgpu/src/data/webgpu-schema.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 Unfolded Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef LUMAGL_WEBGPU_DATA_WEBGPU_SCHEMA_H
+#define LUMAGL_WEBGPU_DATA_WEBGPU_SCHEMA_H
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "./webgpu-field.h"
+
+namespace lumagl {
+
+/// \brief Sequence of WebGPUField objects describing the columns of a table data structure.
+class WebGPUSchema {
+ public:
+  explicit WebGPUSchema(const std::vector<std::shared_ptr<WebGPUField>>& fields) : _fields{fields} {}
+
+  /// \brief Returns number of fields that this schema contains.
```

Comment:
My initial idea was to make this "API compatible" with Arrow https://arrow.apache.org/docs/cpp/api/datatype.html#classarrow_1_1_schema.

Use the same methods, and method naming... 

Right now, this is inspired by schema but offers a slightly different set of methods. Would we loose much by aligning the methods?

It could also mean using PascalCase instead of camelCase in this class - I could go either way on that.